src/formatDate.js

```diff
@@ -8,7 +8,43 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const [toParts] = [toFormat.slice()];
+  const [fromParts] = [fromFormat.slice()];
+  const separators = ['-', '/', ' ', '.'];
+  const separator = findSeparator(date, separators);
+  const newSeparator = findSeparator(toFormat, separators);
```

Comment:
The `findSeparator` function is incorrectly used here. `toFormat` is an array, but `findSeparator` expects a string. You should determine the separator for the output format differently, possibly by using a predefined separator or by ensuring `toFormat` is a string.